### DISCUSSION TITLE
Portable version?

### DISCUSSION THREAD

**Discussion by ghost** [tnodir/fort#71] [Q&A] [answered]
will there a portable version? wanna configure files put in executable path not in Disk C.

**Comment by tnodir** [comment, marked as answer]
You can select "Portable" flag when installing:
![fort-portable](https://user-images.githubusercontent.com/303655/235711671-dc98ba26-f578-4280-a85d-84d40265111e.png)

To Uninstall the Driver: [Functionality-overview: Portability](https://github.com/tnodir/fort/wiki/Functionality-overview#portability).

**Reply by ghost** [reply]
wow ok i will check it later!

**Comment by ghost** [comment]
hi!
i tried portable ver but its still generate cfg to Disk C any ideas??

**Reply by tnodir** [reply]
Because you also selected the “Windows Service” flag.

**Reply by tnodir** [reply]
@surf08 You can just install as non-portable to desired disk and customize paths:
- copy `FortFirewall.exe.example.ini` file to `FortFirewall.exe.ini` in the app's folder
- edit the `profileDir` and `userDir` in the `FortFirewall.exe.ini`

**Reply by tnodir** [reply]
> i tried portable ver but its still generate cfg to Disk C any ideas??

@surf08 Do you mean `%ProgramData%/Fort Firewall`?

I think, Service should use `<APP-PATH>/Data` folder in portable mode too.
Let me check it..

**Reply by ghost** [reply]
Sorry for the late reply, I confirmed that I did not check the other two items, but only checked "portable". The application not only installs the driver itself, but also saves the configuration file to the C drive.  And since this is the "portable" version, without any uninstaller, there are leftovers in the registry after manually removing the drivers and fort folders.

**Reply by tnodir** [reply]
>  also saves the configuration file to the C drive.

Which file do you mean?

> there are leftovers in the registry

Do you mean `HKLM/SOFTWARE/Fort Firewall` in registry?

**Reply by ghost** [reply]
The same configuration files, ini and other files as the installed version.  The registry will leave entries for fort and fort drivers

**Reply by tnodir** [reply]
> The registry will leave entries for fort and fort drivers

Which registry keys do you mean?

I'll double check the portable installation..

**Reply by ghost** [reply]
im sorry im already removed fort, ll check it later

**Comment by ghost** [comment]
I found the problem. It seems that after I deleted 
![reg](https://user-images.githubusercontent.com/105436365/235893843-22887cef-8231-46a8-87b5-4fd03abaced3.PNG)
readme.portable, the configuration file was placed on the c drive. Now as long as I don’t delete it, it’s ok, but the registry problem exists, see the screenshot

**Reply by tnodir** [reply]
The file `README.portable` in the app's folder indicates that installation is portable, so don't remove it.

**Reply by tnodir** [reply]
> but the registry problem exists, see the screenshot

Fort's installed driver can report system events on errors.
So it has an event message source for `Event Viewer` program.

That registry key will be auto-removed after driver uninstalling.

**Comment by ghost** [comment]
By the way, I am looking forward to the launch of the ask for connect function. I can use fort instead of simplewall. Now the mobile hotspot function cannot be used, which makes me very distressed.👍

**Comment by tnodir** [comment]
> Because you also selected the “Windows Service” flag.

Sorry for misinformation, you can use the “Windows Service” flag with "Portable".
